File: Assets/lockpick/MotherScript.js
#pragma strict

static var good : int = 0;
static var reset : boolean = false;
static var now : int = 0;
static var order = new Array (3,5,1,2,4);
static var whichDone = new Array(5);

function Start(){
	for (var i = 0; i < whichDone.length; ++i) { whichDone[i] = false; }
	Shuffle(order);
}

function Update () {
	if(good==5){
		guiText.text='Brawo, udało Ci się!';
	}
	if(good<=order.length-1){
		now=order[good];
	}
	//Debug.Log("Good:"+good);
	//Debug.Log("ktore(NOW)"+now);
	Debug.Log('kolenosc reset'+order);
	//Debug.Log('WHICHdone'+whichDone);
}

function Shuffle(deck: Array) {
    for (var i : int = 0; i < deck.length; i++) {
        var temp = deck[i];
        var randomIndex = Random.Range(0, deck.length);
        deck[i] = deck[randomIndex];
        deck[randomIndex] = temp;
    }
}

File: Assets/lockpick/MotherScript.cs
using UnityEngine;
using System.Collections;

public class MotherScript : MonoBehaviour {
	
	
	static public int good = 0;
	static bool  reset = false;
	static public int now = 0;
	static int[] order={3,5,1,2,4};
	static public bool[] whichDone= new bool[5];
	
	void  Start (){
		for (int i= 0; i < whichDone.Length; ++i) { whichDone[i] = false; }
		Shuffle(order);
	}
	
	void  Update (){
		if(good==5){
			//guiText.text="Brawo, udało Ci się!";
		}
		if(good<=order.Length-1){
			now=order[good];
		}
		Debug.Log("Good:"+good);
		Debug.Log("ktore(NOW)"+now);
	}
	
	void  Shuffle (int[] deck){
		for (int i = 0; i < deck.Length; i++) {
			int temp = deck[i];
			int randomIndex = Random.Range(0, deck.Length);
			deck[i] = deck[randomIndex];
			deck[randomIndex] = temp;
		}
	}
}